src/processor/tx-monitor-queue-consumer.service.ts

```diff
@@ -0,0 +1,149 @@
+import { Injectable, Logger, OnApplicationBootstrap, OnModuleDestroy } from '@nestjs/common';
+import { InjectQueue, Processor, WorkerHost } from '@nestjs/bullmq';
+import { DelayedError, Job, Queue } from 'bullmq';
+import { GraphUpdateJobState, IGraphUpdateJob } from '#app/interfaces/graph-update-job.interface';
+import { BlockchainService } from '#app/blockchain/blockchain.service';
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { BlockchainConstants } from '#app/blockchain/blockchain-constants';
+import { ReconnectionCacheMgrService } from '#app/cache/reconnection-cache-mgr.service';
+import { ReconnectionServiceConstants } from '#app/constants';
+import { ITxStatus } from '#app/interfaces/tx-status.interface';
+import { Graph } from '@dsnp/graph-sdk';
+
+const CAPACITY_EPOCH_TIMEOUT_NAME = 'capacity_check';
+
+@Injectable()
+@Processor(ReconnectionServiceConstants.TX_MONITOR_QUEUE_NAME)
+export class TxMonitorQueueConsumerService extends WorkerHost implements OnApplicationBootstrap, OnModuleDestroy {
```

Comment:
This service monitors the transaction status job queue. These jobs are children of the primary graph update jobs. Each job checks the transaction statuses in redis. If all transactions are accounted for for the parent job, the child job completes successfully and updates the parent job status for its eventual completion. If transactions are still pending, the child job is sent back to the queue with a delay to wait for more blocks.